Sources/Holocron/FileSystemStorageProvider.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  FileSystemStorageProvider.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/15/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+import Foundation
+
+// MARK: Base Struct
+public extension StorageProvider {
+    typealias FileSystem = FileSystemStorageProvider
+}
+
+public struct FileSystemStorageProvider {
+    let baseURL: URL
+    let readTransformer: Transformer
+    let writeTransformer: Transformer
+    
+    public init(baseURL: URL, readTransformer: @escaping Transformer = { $0 }, writeTransformer: @escaping Transformer = { $0 }) {
+        self.baseURL = baseURL
+        self.readTransformer = readTransformer
+        self.writeTransformer = writeTransformer
+    }
+    
+    func url(for key: Key) -> URL {
+        return baseURL.appendingPathComponent(key.rawValue)
+    }
+    
+    public typealias Transformer = (Data) -> Data
```

Comment:
I like placing `typealias` declarations at the top of a struct or class. Not sure if it is a best practice or not.
```
public struct FileSystemStorageProvider {
    public typealias Transformer = (Data) -> Data

    let baseURL: URL
    let readTransformer: Transformer
    let writeTransformer: Transformer
```